src/app.ts

```diff
@@ -0,0 +1,48 @@
+import Vue from 'vue';
+import { sync } from 'vuex-router-sync';
+import BootstrapVue from 'bootstrap-vue';
+import axios from 'axios';
+import { createProxy } from 'vuex-class-component';
+import App from './App.vue';
+import createRouter from './router';
+import createStore from './store/store';
+import InitGlobalComponents from '@/components/global';
+import UserStore from './store/user.vuex';
+import MobileStore from './store/mobile.vuex';
+import PageData from './store/page-data.vuex';
+
+Vue.use(BootstrapVue);
+
+InitGlobalComponents();
+
+Vue.config.productionTip = false;
+
+export default function createApp() {
+  const router = createRouter();
+  const store = createStore();
+
+  // Types for properties added to Vue (prototype) are in types/vue.d.ts
+  // Make sure to update that if you change what is included here!
+
+  const axiosInstance = axios.create({
+    withCredentials: true,
+  });
+  Vue.$http = axiosInstance;
+  Vue.prototype.$http = axiosInstance;
```

Comment:
Why not `$axios`?
If it's for the sake of conventions / being generic, I'd say I'm against it, as this behaves differently from the vue-resource `$http` object, even though they serve the same purpose.

Reply:
I am doing this for the sake of convention - from my understanding, in Vue (and other libraries like Angular) whatever service you're using for networking is located at `$http`. I wouldn't be torn up if we used `$axios`, but I definitely wanted to make sure it was prefixed with `$` (rather than before where it was just `axios`)

Reply:
From the vue-axios examples page:
```
this.axios.get(api).then((response) => {
  console.log(response.data)
})

this.$http.get(api).then((response) => {
  console.log(response.data)
})
```
Looks like I misremembered how it was there; So it seems like it is a bigger convention than I thought.